src/page/post_view/video_loader.rs

```diff
@@ -116,6 +120,69 @@ pub fn VideoView(idx: usize, muted: RwSignal<bool>) -> impl IntoView {
         Some(())
     });
 
+    let watched_percentage_signal = create_rw_signal(0_u8);
+    let watched_count_signal = create_rw_signal(0_u8);
+
+    let mut video_previous_current_time = 0.0;
+    let _ = use_event_listener(container_ref, ev::timeupdate, move |_event| {
+        if let Some(video) = container_ref() {
+            let duration = video.duration();
+            let current_time = video.current_time();
+
+            if current_time < video_previous_current_time {
+                watched_count_signal.update(|watched_count| *watched_count += 1);
+            }
+            watched_percentage_signal.update(|watched_percentage| {
+                *watched_percentage = (100.0 * (current_time / duration)) as u8;
+            });
+            video_previous_current_time = current_time;
+        }
+    });
+
+    let post = video_queue.with_untracked(|vq| vq.get(idx).cloned());
+    let canister_id = post.as_ref().map(|post| post.canister_id).unwrap();
+    let post_id = post.as_ref().map(|post| post.post_id).unwrap();
```

Comment:
these should be functions

Reply:
I think just to fetch data they don't need to necessarily functions.